subprojects/docs/src/docs/userguide/what_is_gradle.adoc

```diff
@@ -12,133 +12,88 @@
-
-Another important aspect of the configuration phase is that everything involved in it is evaluated _every time the build runs_. That is why it's best practice to <<authoring_maintainable_build_scripts#sec:minimize_logic_executed_configuration_phase,avoid expensive work during the configuration phase>>. https://scans.gradle.com/[Build scans] can help you identify such hotspots, among other things.
-
-=== 4. Gradle is extensible in more ways than one
-
-It would be great if you could build your project using only the build logic bundled with Gradle, but that's rarely possible. Most builds have some special requirements that mean you need to add custom build logic.
-
-Gradle provides several mechanisms that allow you to extend it, such as:
-
- * <<custom_tasks#custom_tasks,Custom task types>>.
-+
-When you want the build to do some work that an existing task can't do, you can simply write your own task type. It's typically best to put the source file for a custom task type in the <<organizing_gradle_projects#sec:build_sources,_buildSrc_>> directory or in a packaged plugin. Then you can use the custom task type just like any of the Gradle-provided ones.
-
- * Custom task actions.
-+
-You can attach custom build logic that executes before or after a task via the link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doFirst(org.gradle.api.Action)[Task.doFirst()] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doLast(org.gradle.api.Action)[Task.doLast()] methods.
-
- * <<writing_build_scripts#sec:extra_properties,Extra properties>> on projects and tasks.
-+
-These allows you to add your own properties to a project or task that you can then use from your own custom actions or any other build logic. Extra properties can even be applied to tasks that aren't explicitly created by you, such as those created by Gradle's core plugins.
-
- * Custom conventions.
-+
-Conventions are a powerful way to simplify builds so that users can understand and use them more easily. This can be seen with builds that use standard project structures and naming conventions, such as <<building_java_projects#building_java_projects,Java builds>>. You can write your own plugins that provide conventions — they just need to configure default values for the relevant aspects of a build.
-
- * <<implementing_gradle_plugins.adoc#modeling_dsl_like_apis,A custom model>>.
-+
-Gradle allows you to introduce new concepts into a build beyond tasks, files and dependency configurations. You can see this with most language plugins, which add the concept of <<building_java_projects#sec:java_source_sets,_source sets_>> to a build. Appropriate modeling of a build process can greatly improve a build's ease of use and its efficiency.
```

Comment:
I like this too. is it going somewhere else? T

Reply:
Added this into the plugins section!